src/genie/libs/parser/linux/ps.py

```diff
@@ -77,7 +77,7 @@ def cli(self, output=None, grep=None):
                 del groups['pid']
                 parsed_dict.setdefault('pid', {}).setdefault(pid, groups)
 
-        if len(parsed_dict) == 0:
-            parsed_dict.setdefault('pid', {})
+        #if len(parsed_dict) == 0:

```

Comment:
This did not seem in line with the hundreds of other parsers and would seem to be not as expected to a user of the parsers. That being said, fine to revert if this one-off is as expected.